driver/kubedriver.go

```diff
@@ -323,46 +313,54 @@ func (kd *KubeDriver) upgrade(chartDir string, valueOpts values.Options, group s
 	signal.Notify(cSignal, os.Interrupt, syscall.SIGTERM)
 	go func() {
 		<-cSignal
-		fmt.Printf("experiment for group %s has been cancelled.\n", group)
+		log.Logger.Warnf("experiment for group %s has been cancelled.\n", group)
 		cancel()
 	}()
 
-	rel, err := client.RunWithContext(ctx, group, ch, vals)
+	rel, err := client.RunWithContext(ctx, group, chartRequested, vals)
 	if err != nil {
 		e := fmt.Errorf("experiment launch failed")
 		log.Logger.WithStackTrace(err.Error()).Error(e)
 		return e
 	}
 
-	// upgrading revision info
-	kd.revision = rel.Version
-
-	// write manifest if dry
-	if dry {
-		err := writeManifest(rel)
-		if err != nil {
-			return err
-		}
-		log.Logger.Info("dry run complete")
-	} else {
-		log.Logger.Info("experiment launched. Happy Iter8ing!")
-	}
-
-	return nil
+	return kd.updateRevision(rel, dry)
 }
 
 // install a Kubernetes experiment
 // Credit: the logic for this function is sourced from Helm
 // https://github.com/helm/helm/blob/8ab18f7567cedffdfa5ba4d7f6abfb58efc313f8/cmd/helm/install.go#L177
-func (kd *KubeDriver) install(chartDir string, valueOpts values.Options, group string, dry bool) error {
+func (kd *KubeDriver) install(chartPathOptions action.ChartPathOptions, chartName string, valueOpts values.Options, group string, dry bool) error {
+
+	// buf := new(bytes.Buffer)
 	client := action.NewInstall(kd.Configuration)
 	client.Namespace = kd.Namespace()
+	client.RepoURL = chartPathOptions.RepoURL
 	client.DryRun = dry
 	client.ReleaseName = group
 
-	ch, vals, err := kd.getChartAndVals(chartDir, valueOpts)
+	// copy chartPathOptions to client
+	client.ChartPathOptions.CaFile = chartPathOptions.CaFile
```

Comment:
Could we please remove all of this? We don't even plan to document how the user is intended to use any of these options ... for now, we are better of without this.

We do need to discuss binary/image/chart signing and verification. We will revisit all of these options at that time.

Reply:
This code is initializing a field (the `ChartPathOptions` field) in the Helm object (`client`). This object is used both to find the chart and to call `RunWithContext()` to do the actual work. We do want to initialize the field. However, we may not need to copy every field. I can comment out the fields that we do not explicitly set.